subprojects/launcher/src/integTest/groovy/org/gradle/launcher/cli/CommandLineIntegrationLoggingSpec.groovy

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.cli
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import spock.lang.Ignore
+import spock.lang.Unroll
+
+class CommandLineIntegrationLoggingSpec extends AbstractIntegrationSpec {
+
+    @Unroll
+    def "Set logging level using org.gradle.logging.level=#logLevel"() {
+        def message = 'Expected message in the output'
+        buildFile << """
+            task assertLogging {
+                doLast {
+                    logger.${logLevel} '${message}'
+                    assert logger.${logLevel}Enabled
+                    if ('${nextLevel}') { // check that there is a next level (there isn't in DEBUG)
+                        assert !logger.${nextLevel}Enabled
+                    }
+                }
+            }
+        """
+        expect:
+        executer.withArguments(flags)
+        succeeds("assertLogging")
+        outputContains(message)
+
+        where:
+        logLevel    | nextLevel   | flags
+        'quiet'     | 'warn'      | ['-Dorg.gradle.logging.level=quiet']
+        'warn'      | 'lifecycle' | ['-Dorg.gradle.logging.level=warn']
+        'lifecycle' | 'info'      | ['-Dorg.gradle.logging.level=lifecycle']
+        'info'      | 'debug'     | ['-Dorg.gradle.logging.level=info']
+        'debug'     | ''          | ['-Dorg.gradle.logging.level=debug']
+    }
+
+    @Ignore("Could not get this test to work, even when extending DaemonIntegrationSpec")
```

Comment:
Hmmm, I got that to work with the existing test if I am not mistaken. I can check again later after merging my branch and trying it out on yours.